### DISCUSSION TITLE
'weights' argument not passed to the function 'multi_source_dijkstra_path' in Mehlhorn Steiner algorithm

### DISCUSSION THREAD

**Discussion by mm2801** [networkx/networkx#7630] [Q&A] [answered]
Hi,
why aren't the weights passed to the function multi_source_dijkstra_path?
Shouldn't it lead to a better approximated Steiner tree if the shortest path determination in the function takes the weights into account?

**Comment by dschult** [comment, marked as answer]
The Mehlhorn paper says the distance function should be the sum of the edge distances. So when a weight is specified to the `steiner_tree` function with the `mehlhorn` method it looks to me like the `weight` parameter [should be used to compute the shortest paths](https://github.com/networkx/networkx/blob/409979eff35f02eff54f4eea3731736bd431dc2e/networkx/algorithms/approximation/steinertree.py#L51).  Would you like to contribute a fix for this? (or we can do it)

Thanks for this!

**Reply by mm2801** [reply]
I would be happy if you could do that, as it is just a small adjustment.
